protocol-impl/src/main/java/io/zeebe/protocol/impl/record/RecordMetadata.java

```diff
@@ -64,6 +73,16 @@ public void wrap(final DirectBuffer buffer, int offset, final int length) {
     intent = Intent.fromProtocolValue(valueType, decoder.intent());
     rejectionType = decoder.rejectionType();
 
+    brokerVersion =
+        Optional.ofNullable(decoder.brokerVersion())
+            .map(
+                versionDecoder ->
+                    new VersionInfo(
+                        versionDecoder.majorVersion(),
+                        versionDecoder.minorVersion(),
+                        versionDecoder.patchVersion()))
+            .orElse(VersionInfo.UNKNOWN);
```

Comment:
If it reads a record from a previous version then `decoder.brokerVersion()` returns `null`.